query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs

```diff
@@ -24,28 +24,21 @@ pub async fn create_record<'conn>(
     mut args: WriteArgs,
 ) -> crate::Result<SelectionResult> {
     let coll = database.collection::<Document>(model.db_name());
-
-    // Mongo only allows a singular ID.
-    let mut id_fields = ModelProjection::from(model.primary_identifier())
-        .scalar_fields()
-        .collect::<Vec<_>>();
-
-    assert!(id_fields.len() == 1);
-
-    let id_field = id_fields.pop().unwrap();
+    let id_field = pick_singular_id(model);
 
     // Fields to write to the document.
     // Todo: Do we need to write null for everything? There's something with nulls and exists that might impact
     //       query capability (e.g. query for field: null may need to check for exist as well?)
     let fields: Vec<_> = model
         .fields()
-        .scalar()
-        .into_iter()
+        .all
```

Comment:
The mongo internals are not finished yet, a wider cleanup is indeed on the radar already but not in scope here, also because we're not 100% sure how it will look like in the end so we're not doing what we consider premature refactoring here. This applies to most of your requests like this.